client/src/components/UploadComponent.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from "react";
+import { v4 as uuidv4 } from "uuid";
+
+import { useHttpClient } from "../hooks/http-hook";
+
+const UploadComponent = ({ addVideo }) => {
+  const randomId = uuidv4();
+  const [videoData, setVideoData] = useState({
+    id: randomId,
+    url: "",
+    title: "",
+    rating: 0,
+  });
+  const { sendRequest, isLoading, error } = useHttpClient();
+
+  const handleChange = ({ target }) => {
+    const key = target.placeholder.toLowerCase();
+    setVideoData({ ...videoData, [key]: target.value });
+  };
+
+  const handleAdding = async () => {
+    try {
+      if (!videoData.url.trim() && !videoData.title.trim() && !videoData.id) {
+        throw new Error("Video URL and title are required.");
+      }
+      const response = await sendRequest(
+        `${process.env.REACT_APP_API_URL}/`,
+        "POST",
+        JSON.stringify(videoData),
+        {
+          "Content-Type": "application/json",
+        }
+      );
+      addVideo(videoData);
+      setVideoData({
+        id: randomId,
+        url: "",
+        title: "",
+        rating: 0,
+      });
+      console.log(response);
+    } catch (err) {
+      console.log("Error adding video.", { error: err });
```

Comment:
A user will never see this error - how can you ensure the user actually knows when something went wrong, and what they should do about it?